arlo_server/jurisdictions.py

```diff
@@ -1,35 +1,139 @@
 from flask import jsonify
+from typing import List, Dict, Optional
+from enum import Enum
+from sqlalchemy import func
 
 from arlo_server import app
-from arlo_server.models import Election, Jurisdiction
+from arlo_server.models import (
+    Election,
+    Jurisdiction,
+    SampledBallotDraw,
+    SampledBallot,
+    Batch,
+    Round,
+    AuditBoard,
+)
 from arlo_server.auth import with_election_access, UserType
+from arlo_server.rounds import get_current_round
 from util.process_file import serialize_file, serialize_file_processing
+from util.jsonschema import JSONDict
 
 
-def serialize_jurisdiction(db_jurisdiction: Jurisdiction) -> dict:
+def serialize_jurisdiction(
+    jurisdiction: Jurisdiction, round_status: Optional[JSONDict]
+) -> JSONDict:
     return {
-        "id": db_jurisdiction.id,
-        "name": db_jurisdiction.name,
+        "id": jurisdiction.id,
+        "name": jurisdiction.name,
         "ballotManifest": {
-            "file": serialize_file(db_jurisdiction.manifest_file)
-            if db_jurisdiction.manifest_file
+            "file": serialize_file(jurisdiction.manifest_file)
+            if jurisdiction.manifest_file
             else None,
-            "processing": serialize_file_processing(db_jurisdiction.manifest_file)
-            if db_jurisdiction.manifest_file
+            "processing": serialize_file_processing(jurisdiction.manifest_file)
+            if jurisdiction.manifest_file
             else None,
-            "numBallots": db_jurisdiction.manifest_num_ballots,
-            "numBatches": db_jurisdiction.manifest_num_batches,
+            "numBallots": jurisdiction.manifest_num_ballots,
+            "numBatches": jurisdiction.manifest_num_batches,
         },
+        "currentRoundStatus": round_status,
+    }
+
+
+class JurisdictionStatus(str, Enum):
+    NOT_STARTED = "NOT_STARTED"
+    IN_PROGRESS = "IN_PROGRESS"
+    COMPLETE = "COMPLETE"
+
+
+def round_status_by_jurisdiction(
+    round: Optional[Round], jurisdictions: List[Jurisdiction], online: bool
+) -> Dict[str, Optional[JSONDict]]:
+    if not round:
+        return {j.id: None for j in jurisdictions}
+
+    audit_boards_set_up = dict(
+        AuditBoard.query.filter_by(round_id=round.id)
+        .group_by(AuditBoard.jurisdiction_id)
+        .values(AuditBoard.jurisdiction_id, func.count())
+    )
+    audit_boards_signed_off = dict(
+        AuditBoard.query.filter_by(round_id=round.id)
+        .filter(AuditBoard.signed_off_at.isnot(None))
+        .group_by(AuditBoard.jurisdiction_id)
+        .values(AuditBoard.jurisdiction_id, func.count())
+    )
```

Comment:
fyi, you could do this in one query with two `count` functions, one that is `count(*)` and one that is `count(signed_off_at)`, see: https://stackoverflow.com/questions/37328779/sqlalchemy-func-count-on-boolean-column

Reply:
Yeah, I was playing around with doing it in one query. Two reason I ended up with this:
- I like the way the code reads when you return tuples and turn them right into a dict.
- I was having a hard time making it actually give me the right answer in a way that I felt confident about. I think I was using something like `.values(AuditBoard.jurisdiction_id, func.count(), func.count().filter(AuditBoard.signed_off_at.isnot(None))` and it seemed like the filter was applying to the whole query and not just the count. I'm sure there was a way to do it, but I decided to bail since it seemed like even if I figured it out, it would be brittle/easy to break if we ever tried to change it.

The only advantage was saving one db query which didn't seem worthwhile at this time.